src/packageurl/contrib/purl2url.py

```diff
@@ -304,6 +310,25 @@ def build_golang_repo_url(purl):
         return f"https://pkg.go.dev/{namespace}/{name}"
 
 
+@repo_router.route("pkg:cocoapods/.*")
+def build_cocoapods_repo_url(purl):
+    """
+    Return a CocoaPods repo URL from the `purl` string.
+    """
+    purl_data = None
+    name = None
+    try:
```

Comment:
Same as above: if the purl is invalid, let the error bubble up and do not catch the exception.

Reply:
@pombredanne I'm not sure I understand your comments but hoping this is what you suggested.  I've removed the try/except as well as this if condition:

```
    if not name:
        return
```

 and the updated `build_cocoapods_repo_url()` looks like this:

```
@repo_router.route("pkg:cocoapods/.*")
def build_cocoapods_repo_url(purl):
    """
    Return a CocoaPods repo URL from the `purl` string.
    """
    purl_data = PackageURL.from_string(purl)
    name = purl_data.name
    return name and f"https://cocoapods.org/pods/{name}"
```

I also removed `"pkg:cocoapods/": None,` from `test_purl2url_get_repo_url()` because, having removed the try/except, that test fails on `"pkg:cocoapods/": None,` (which I'd added only to test the try/except).